tasks/pkg-windows.yml

```diff
@@ -4,6 +4,31 @@
     msg: "The Datadog ansible role does not currently support Agent 5"
   when: agent_datadog_agent_major_version|int == 5
 
+## must be prior to `pkg-windows-opts.yml`, because the variable is used inside
+- name: Set windows NPM installed
+  set_fact:
+    agent_datadog_sysprobe_enabled: "{{ network_config is defined and 'enabled' in (network_config | default({}, true)) and network_config['enabled'] }}"
+
+- name: Include Windows opts tasks
+  include_tasks: pkg-windows-opts.yml
+
+- name: Show installation flags
+  debug:
+    msg: "{{ agent_win_install_args }}{% if datadog_windows_ddagentuser_password | default('', true) | length > 0 %} DDAGENTUSER_PASSWORD=<REDACTED>{% endif %}"
+
+- name: Collect previous agent install arguments
+  win_reg_stat:
+    path: HKLM:\SOFTWARE\Datadog\Datadog Agent
+    name: InstallArgs
+  register: agent_config_install_args
+
+- name: Force install if config changed
```

Comment:
I think you would also need to add previous value of agent_datadog_skip_install in the condition:
```
{{ agent_datadog_skip_install and agent_config_install_args.exists and
      agent_config_install_args.value != agent_win_install_args }}
```
Otherwise we could skip the install when we want to install another version of the agent with the same install args